cluster/charts/rook-ceph/templates/deployment.yaml

```diff
@@ -116,109 +116,71 @@ spec:
         - name: CSI_CEPHFS_PLUGIN_UPDATE_STRATEGY
           value: {{ .Values.csi.cephFSPluginUpdateStrategy | quote }}
 {{- end }}
-{{- if .Values.csi.rbdPluginUpdateStrategy }}
         - name: CSI_RBD_PLUGIN_UPDATE_STRATEGY
           value: {{ .Values.csi.rbdPluginUpdateStrategy | quote }}
-{{- end }}
-{{- if .Values.csi.kubeletDirPath }}
         - name: ROOK_CSI_KUBELET_DIR_PATH
           value: {{ .Values.csi.kubeletDirPath | quote }}
-{{- end }}
         - name: ROOK_CSI_ENABLE_GRPC_METRICS
           value: {{ .Values.csi.enableGrpcMetrics | quote }}
-{{- if .Values.csi.cephcsi }}
-{{- if .Values.csi.cephcsi.image }}
```

Comment:
The expected approach for the CSI settings is that Rook has built-in defaults in the code. The settings should only be applied in the helm chart (or operator.yaml) if the non-defaults are desired by the admin. We don't want to set the defaults in the helm chart or else the upgrade process is not as smooth. Can you explain more why this change was made for the new helm?